core/src/common/cdk-deployer-build.ts

```diff
@@ -0,0 +1,94 @@
me,\n      // Pass CFN related parameters through the build for extraction by the\n      // completion handler.\n      buildspecOverride: event.RequestType === 'Delete' ? \n      `\nversion: 0.2\nenv:\n  variables:\n    CFN_RESPONSE_URL: CFN_RESPONSE_URL_NOT_SET\n    CFN_STACK_ID: CFN_STACK_ID_NOT_SET\n    CFN_REQUEST_ID: CFN_REQUEST_ID_NOT_SET\n    CFN_LOGICAL_RESOURCE_ID: CFN_LOGICAL_RESOURCE_ID_NOT_SET\nphases:\n  pre_build:\n    on-failure: ABORT\n    commands:\n      - cd $CODEBUILD_SRC_DIR/$CDK_APP_LOCATION\n      - npm install -g aws-cdk && sudo apt-get install python3 && python -m\n        ensurepip --upgrade && python -m pip install --upgrade pip && python -m\n        pip install -r requirements.txt\n      - \"export AWS_ACCOUNT_ID=$(echo $CODEBUILD_BUILD_ARN | cut -d: -f5)\"\n      - 'echo \"AWS_ACCOUNT_ID: $AWS_ACCOUNT_ID\"'\n      - cdk bootstrap aws://$AWS_ACCOUNT_ID/$AWS_REGION\n  build:\n    on-failure: ABORT\n    commands:\n      - \"export AWS_ACCOUNT_ID=$(echo $CODEBUILD_BUILD_ARN | cut -d: -f5)\"\n      - 'echo \"AWS_ACCOUNT_ID: $AWS_ACCOUNT_ID\"'\n      - cdk destroy --force --all --require-approval never\n      `\n      :\n      `\nversion: 0.2\nenv:\n  variables:\n    CFN_RESPONSE_URL: CFN_RESPONSE_URL_NOT_SET\n    CFN_STACK_ID: CFN_STACK_ID_NOT_SET\n    CFN_REQUEST_ID: CFN_REQUEST_ID_NOT_SET\n    CFN_LOGICAL_RESOURCE_ID: CFN_LOGICAL_RESOURCE_ID_NOT_SET\n    PARAMETERS: PARAMETERS_NOT_SET\n    STACKNAME: STACKNAME_NOT_SET\nphases:\n  pre_build:\n    on-failure: ABORT\n    commands:\n      - cd $CODEBUILD_SRC_DIR/$CDK_APP_LOCATION\n      - npm install -g aws-cdk && sudo apt-get install python3 && python -m\n        ensurepip --upgrade && python -m pip install --upgrade pip && python -m\n        pip install -r requirements.txt\n      - \"export AWS_ACCOUNT_ID=$(echo $CODEBUILD_BUILD_ARN | cut -d: -f5)\"\n      - 'echo \"AWS_ACCOUNT_ID: $AWS_ACCOUNT_ID\"'\n      - cdk bootstrap aws://$AWS_ACCOUNT_ID/$AWS_REGION\n  build:\n    on-failure: ABORT\n    commands:\n      - \"export AWS_ACCOUNT_ID=$(echo $CODEBUILD_BUILD_ARN | cut -d: -f5)\"\n      - 'echo \"AWS_ACCOUNT_ID: $AWS_ACCOUNT_ID\"'\n      - cdk deploy $STACKNAME $PARAMETERS --require-approval=never\n      `,\n      environmentVariablesOverride: [\n        {\n          name: 'CFN_RESPONSE_URL',\n          value: event.ResponseURL\n        },\n        {\n          name: 'CFN_STACK_ID',\n          value: event.StackId\n        },\n        {\n          name: 'CFN_REQUEST_ID',\n          value: event.RequestId\n        },\n        {\n          name: 'CFN_LOGICAL_RESOURCE_ID',\n          value: event.LogicalResourceId\n        },\n        {\n          name: 'BUILD_ROLE_ARN',\n          value: event.ResourceProperties.BuildRoleArn\n        }\n      ]\n    }).promise();\n    console.log(`Build id ${build.id} started - resource completion handled by EventBridge`);\n  } catch(error) {\n    console.error(error);\n    await respond(event, context, 'FAILED', { Error: error });\n  }\n};"
```

Comment:
I would suggest to put in multi line and avoid overriding buildspec because here you are overriding whatever you put in the other construct. can do in my PR